tests/test_lib_utils.py

```diff
@@ -781,3 +781,26 @@ def test_32_allowed_serial_numbers(self):
 
         # an empty serial is not allowed
         self.assertRaises(Exception, check_serial_valid, "")
+
+    def test_33_determine_logged_in_user(self):
+        (scope, user, realm, adminuser, adminrealm) = determine_logged_in_userparams({"role": "user",
+                                                                                      "username": "hans",
+                                                                                      "realm": "realm1"}, {})
+
+        self.assertEqual(scope, "user")
+        self.assertEqual(user, "hans")
+        self.assertEqual(realm, "realm1")
+        self.assertEqual(adminuser, None)
+        self.assertEqual(adminrealm, None)
+
+        (scope, user, realm, adminuser, adminrealm) = determine_logged_in_userparams({"role": "admin",
+                                                                                      "username": "hans",
+                                                                                      "realm": "realm1"},
+                                                                                     {"user": "peter",
+                                                                                      "realm": "domain"})
+
+        self.assertEqual(scope, "admin")
+        self.assertEqual(user, "peter")
+        self.assertEqual(realm, "domain")
+        self.assertEqual(adminuser, "hans")
+        self.assertEqual(adminrealm, "realm1")
```

Comment:
Please also check for the PolicyError in case the "role" is unknown.

Reply:
will check and do.

Reply:
Done in ff3bb82be7f1697a96015bfa486b034c659d3e0f